src/System.Security.Permissions/src/System/Security/SecurityElement.cs

```diff
@@ -555,129 +543,82 @@ private static string Unescape(string str)
         }
 
-        private void ToString(string indent, object obj, ToStringHelperFunc func)
+        private void ToString(string indent, object obj, Action<object, string> write)
         {
-            func(obj, "<");
-
-            switch (m_type)
-            {
-                case SecurityElementType.Format:
-                    func(obj, "?");
-                    break;
-
-                case SecurityElementType.Comment:
-                    func(obj, "!");
-                    break;
-
-                default:
-                    break;
-            }
-
-            func(obj, m_strTag);
+            write(obj, "<");
+            write(obj, _tag);
 
             // If there are any attributes, plop those in.
-            if (m_lAttributes != null && m_lAttributes.Count > 0)
+            if (_attributes != null && _attributes.Count > 0)
             {
-                func(obj, " ");
+                write(obj, " ");
 
-                int iMax = m_lAttributes.Count;
+                int iMax = _attributes.Count;
                 Debug.Assert(iMax % 2 == 0, "Odd number of strings means the attr/value pairs were not added correctly");
 
                 for (int i = 0; i < iMax; i += 2)
                 {
-                    string strAttrName = (string)m_lAttributes[i];
-                    string strAttrValue = (string)m_lAttributes[i + 1];
+                    string strAttrName = (string)_attributes[i];
+                    string strAttrValue = (string)_attributes[i + 1];
 
-                    func(obj, strAttrName);
-                    func(obj, "=\"");
-                    func(obj, strAttrValue);
-                    func(obj, "\"");
+                    write(obj, strAttrName);
+                    write(obj, "=\"");
+                    write(obj, strAttrValue);
+                    write(obj, "\"");
 
-                    if (i != m_lAttributes.Count - 2)
+                    if (i != _attributes.Count - 2)
                     {
-                        if (m_type == SecurityElementType.Regular)
-                        {
-                            func(obj, Environment.NewLine);
-                        }
-                        else
-                        {
-                            func(obj, " ");
-                        }
+                        write(obj, Environment.NewLine);
                     }
                 }
             }
 
-            if (m_strText == null && (m_lChildren == null || m_lChildren.Count == 0))
+            if (_text == null && (_children == null || _children.Count == 0))
             {
                 // If we are a single tag with no children, just add the end of tag text.
-                switch (m_type)
-                {
-                    case SecurityElementType.Comment:
-                        func(obj, ">");
-                        break;
-
-                    case SecurityElementType.Format:
-                        func(obj, " ?>");
```

Comment:
The `m_type` variable was an internal readonly that was hardcoded to SecurityElementType.Regular, so this was dead code.

It may need to be re-added when/if we flesh out the types that use SecurityElement, but until that time